plugins/woocommerce/client/legacy/js/frontend/address-autocomplete.js

```diff
@@ -738,6 +748,9 @@ window.wc.addressAutocomplete.registerAddressAutocompleteProvider =
 						addressInputs[ type ][ 'address_2' ],
 						addressData.address_2
 					);
+				} else {
+					// Clear address_2 if not provided in address data
+					setFieldValue( addressInputs[ type ][ 'address_2' ], '' );
```

Comment:
I don't see anywhere in this function where we're checking that `addressInputs[ type ]` exists. I think it would be good to add some defensive checks to ensure `type` exists on the object. For example I can see that in `displaySuggestions` we do ensure we're checking for the existence before access a value on that object but not so much in this function.

Reply:
There are other usages of `setFieldValue` in this file that assume the object property exists. I think we can make more robust changes there too at the same time with defensive checks.